### PR TITLE
FAST plugin system

### LINKED ISSUE #1262: FAST plugin system

### PR DESCRIPTION
**Pull request by ludoo**
WIP

Fixes #1262

### CODE REVIEW

**Comment by ludoo** [comment]
@juliocc how do we test these, other than by hand when developing? :)

**Comment by ludoo** [comment]
> Beautiful. I really like it.
> 
> > @juliocc how do we test these, other than by hand when developing? :)
> 
> There's no reason we can't test them as a regular module. Just drop a few examples in the readme
> 
> For a more exhaustive test, we can try to enable them before running the corresponding FAST stage. I'll have to think about it a bit more.

Plugins depend on the stage internals, we need to test them as part of their stage. Any ideas on how to do that in a simple way?

File: tests/fast/stages/s2_networking_a_peering/tftest.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+
+module: fast/stages/2-networking-a-peering
+common_tfvars:
+  - common.tfvars
+
+tests:
+  stage:
+    extra_files:
+      - ../../plugins/2-networking-serverless-connector/*.tf
```
Comment (juliocc) [inline tests/fast/stages/s2_networking_a_peering/tftest.yaml:22]: if you want to test the plugin, you have to set `var.serverless_connector_config` in the tfvars

File: tests/fast/stages/s2_networking_a_peering/tftest.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module: fast/stages/2-networking-a-peering
+common_tfvars:
+  - common.tfvars
+
+tests:
+  stage:
+    extra_files:
```
Comment (juliocc) [inline tests/fast/stages/s2_networking_a_peering/tftest.yaml:21]: do we want to test the plugin in other networking stages too?

**Review by juliocc — CHANGES_REQUESTED** [review]

File: tests/fast/stages/s2_networking_a_peering/tftest.yaml (hunk 1 above)
Comment (ludoo) [reply tests/fast/stages/s2_networking_a_peering/tftest.yaml:22]: not really, the default kicks in

File: tests/fast/stages/s2_networking_a_peering/tftest.yaml (hunk 2 above)
Comment (ludoo) [reply tests/fast/stages/s2_networking_a_peering/tftest.yaml:21]: Ideally yes

File: tests/fast/stages/s2_networking_a_peering/tftest.yaml (hunk 1 above)
Comment (ludoo) [reply tests/fast/stages/s2_networking_a_peering/tftest.yaml:22]: image

File: tests/fast/stages/s2_networking_a_peering/tftest.yaml (hunk 2 above)
Comment (juliocc) [reply tests/fast/stages/s2_networking_a_peering/tftest.yaml:21]: Done. Merge!

**Review by juliocc — APPROVED** [review]

**Pull request merged**